### PR TITLE
Flow verification - "ping"

### PR DESCRIPTION
**Pull request by surabujin**
We can inject "ping"(validation) packet on edge switch, it will be
"routed" using flow rules on all intermediate switches and will be
catched on the edge switch on other side of the flow ("packet out"
message must specify special out port called "TABLE" to pass packet via
existing switch's open flow rules).

This change add the ability to make such ping for some particallar flow.
It can be request via NB API call. All management work is done inside
"flow" topology. Switch and packet manipulation is done on FL side.

There is no obstacles to "call" this feature from some other part of
project (there is a plan to use it into switch discovery/validation
process).

### CODE REVIEW

**Comment by surabujin** [comment]
Depends from https://github.com/kilda/floodlight/pull/4 (it can't be compiled without patch in FL)

**Comment by surabujin** [comment]
CI build fails with
```
[ERROR] Tests run: 27, Failures: 0, Errors: 1, Skipped: 4, Time elapsed: 136.746 s <<< FAILURE! - in org.openkilda.wfm.topology.flow.FlowTopologyTest
[ERROR] shouldSyncCacheWithFlowsTest(org.openkilda.wfm.topology.flow.FlowTopologyTest)  Time elapsed: 33.078 s  <<< ERROR!
java.lang.NullPointerException
	at org.openkilda.wfm.topology.flow.FlowTopologyTest.shouldSyncCacheWithFlowsTest(FlowTopologyTest.java:1079)
```

This is exact same problem that was with build https://github.com/telstra/open-kilda/pull/680 and was "solved" with https://github.com/telstra/open-kilda/pull/680/commits/c20c4876efa42b2b441f7c6227f62ac186712c16

Looks like there is some race condition that hame more chance to happend on CI servers.

PS This issues have no relation to changes from this PR.

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/service/batch/OFPendingMessage.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openkilda.floodlight.service.batch;
+
+import org.projectfloodlight.openflow.protocol.OFMessage;
+import org.projectfloodlight.openflow.types.DatapathId;
+
+public class OFPendingMessage {
```
Comment (nikitamarchenko) [inline services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/service/batch/OFPendingMessage.java]: use lombok

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/KildaModule.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import net.floodlightcontroller.core.module.FloodlightModuleException;
+import net.floodlightcontroller.core.module.IFloodlightModule;
+import net.floodlightcontroller.core.module.IFloodlightService;
+import net.floodlightcontroller.threadpool.IThreadPoolService;
+
+import java.util.Collection;
+import java.util.Map;
+
+public class KildaModule implements IFloodlightModule {
```
Comment (nikitamarchenko) [inline services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/KildaModule.java]: Rename that to clear name, refactoring of FL modules here #697

File: services/src/api/src/main/java/org/openkilda/northbound/dto/flows/UniFlowVerificationOutput.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+package org.openkilda.northbound.dto.flows;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@JsonSerialize
```
Comment (sergii-iakovenko) [inline services/src/api/src/main/java/org/openkilda/northbound/dto/flows/UniFlowVerificationOutput.java]: JsonSerialize annotation is redundant if  a custom serializer or other aspects are not specified.

File: services/src/api/src/main/java/org/openkilda/northbound/dto/flows/UniFlowVerificationOutput.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 Telstra Open Source
```
Comment (sergii-iakovenko) [inline services/src/api/src/main/java/org/openkilda/northbound/dto/flows/UniFlowVerificationOutput.java]: Shouldn't it be 2018?

Comment (surabujin) [reply services/src/api/src/main/java/org/openkilda/northbound/dto/flows/UniFlowVerificationOutput.java]: Good question... I don't know.

**Pull request merged**